### PR TITLE
Fix/rh overwrite

### PR DESCRIPTION
**Pull request by mlindauer**
required for my auto_net implementation

### CODE REVIEW

**Comment by mfeurer** [comment]
Could you please give an intuition on why this is necessary? To re-run failed runs?

**Comment by mlindauer** [comment]
I'm calling the target algorithm (i.e., param_net) multiple time -- each time it continues the optimization of the NN. Since the TAE classes adds each run to the runhistory, the current implementation saves only the first call to the TAE and silently ignores all future calls. This PR fixes this problem.

For example:
1. Call TAE with config_1 and train the NN for 5 epochs; returns loss of 0.5
2. Call TAE with config_1 and train the NN for another 5 epochs; returns loss of 0.3
3. Call TAE with config_1 and train the NN for another 5 epochs; returns loss of 0.2

The runhistory will save only the first results (0.5) and ignores the improved performance values.
However after the 3 calls, the performance of config_1 in the runhistory should be 0.2.

File: smac/runhistory/runhistory.py [hunk 1]
```diff
@@ -49,9 +49,16 @@ class RunHistory(object):
...
+    overwrite_existing: bool
+        allows to overwrites old results if pairs of 
+        algorithm-instance-seed were measured  
+        multiple times
     '''
 
-    def __init__(self, aggregate_func):
+    def __init__(self, 
+                 aggregate_func: typing.Callable,
+                 overwrite_existings: bool=False
```
Comment (KEggensperger) [inline smac/runhistory/runhistory.py]: This should be "overwrite_existing" without the 's' as in the docstring. "overwrite_existing_runs" would be even better

Comment (mlindauer) [inline smac/runhistory/runhistory.py]: Thanks. fixed

**Comment by mlindauer** [comment]
@KEggensperger I see your point. However, the rh is already a memory bottleneck and I'm not convinced that we should store learning curves / all runs in it.

**Pull request merged**